fiftyone/utils/hf_hub.py

```diff
@@ -0,0 +1,1109 @@
+    label_field_names, label_types = [], []
+    label_fields = config.label_fields
+    if label_fields is None:
+        return label_field_names, label_types
+
+    for label_type, fields in label_fields.items():
+        if isinstance(fields, str):
+            label_field_names.append(fields)
+            label_types.append(label_type)
+        elif isinstance(fields, list):
+            label_field_names.extend(fields)
+            label_types.extend([label_type] * len(fields))
+
+    return label_field_names, label_types
+
+
+def _get_parquet_dataset_features(
+    repo_id, split, subset, revision=None, **kwargs
+):
+    api_url = f"{DATASETS_SERVER_URL}/info?dataset={repo_id.replace('/', '%2F')}&config={subset}&split={split}]"
+    if revision is not None:
+        api_url += f"&revision={revision}"
+
+    headers = _get_headers(**kwargs)
+    response = requests.get(api_url, headers=headers)
+    features = response.json()["dataset_info"]["features"]
+    return features
+
+
+def _get_num_rows(repo_id, split, subset, revision=None, **kwargs):
+    api_url = f"{DATASETS_SERVER_URL}/info?dataset={repo_id.replace('/', '%2F')}&config={subset}&split={split}]"
+    if revision is not None:
+        api_url += f"&revision={revision}"
+
+    headers = _get_headers(**kwargs)
+    response = requests.get(api_url, headers=headers)
+    splits = response.json()["dataset_info"]["splits"]
+    return splits[split]["num_examples"]
+
+
+def _build_rows_request_url(
+    repo_id, split=None, subset="default", revision=None, offset=0, length=100
+):
+    url = f"{DATASETS_SERVER_URL}/rows?dataset={repo_id.replace('/', '%2F')}"
+    if split is not None:
+        url += f"&split={split}"
+    if subset is not None:
+        url += f"&config={subset}"
+    if revision is not None:
+        url += f"&revision={revision}"
+    url += f"&offset={offset}&length={length}"
+    return url
+
+
+def _get_rows(
+    repo_id,
+    split,
+    subset,
+    start_index=0,
+    end_index=100,
+    revision=None,
+    **kwargs,
+):
+    length = end_index - start_index
+    url = _build_rows_request_url(
+        repo_id, split, subset, revision, offset=start_index, length=length
+    )
+    headers = _get_headers(**kwargs)
+    response = requests.get(url, headers=headers)
+    return response.json()["rows"]
+
+
+def _download_image(url_and_filepath):
+    url, filepath = url_and_filepath
+    try:
+        if not os.path.exists(filepath):
+            with requests.get(url, stream=True) as r:
+                r.raise_for_status()
+                with open(filepath, "wb") as f:
+                    for chunk in r.iter_content(chunk_size=8192):
+                        f.write(chunk)
+                # Optionally, open and save the image to ensure it's valid
+                Image.open(filepath).save(filepath)
+    except Exception as e:
+        logger.warning(f"Failed to download image from {url}: {e}")
+
+
+def _download_images_in_batch(urls_and_filepaths):
```

Comment:
@jacobmarks can you add a `num_workers` parameter to `load_from_hub()` that gets passed through to here that lets the user configure the thread pool size if desired? As done here, for example:
https://github.com/voxel51/fiftyone/blob/df0b63b64c56bdb3d51d19137d1b182ed8a57431/fiftyone/core/metadata.py#L253

Note that there's a utility to help parse `num_workers` and apply `fo.config` limits if needed:

```py
num_workers = fou.recommend_thread_pool_workers(num_workers)
```

Also, by convention we typically add a separate code path for `num_workers <= 1` that avoids the use of thread pools altogether. It's admittedly less important for threads but this is important when using process pools, because some systems (eg Windows) and some environments (eg when this code runs in a plugin) don't like/support multiprocessing. Here's an example of this:
https://github.com/voxel51/fiftyone/blob/df0b63b64c56bdb3d51d19137d1b182ed8a57431/fiftyone/core/metadata.py#L283C5-L283C25